src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/cache_tuning/dedicated_master/CacheRcaDedicatedMasterITest.java

```diff
@@ -0,0 +1,253 @@
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.integTests.framework.runners.RcaItNotEncryptedRunner;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.integTests.tests.cache_tuning.Constants;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.integTests.tests.cache_tuning.validator.FieldDataCacheValidator;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.integTests.tests.cache_tuning.validator.ShardRequestCacheValidator;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.ElasticSearchAnalysisGraph;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.FieldDataCacheClusterRca;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.ShardRequestCacheClusterRca;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+@Category(RcaItMarker.class)
+@RunWith(RcaItNotEncryptedRunner.class)
+@AClusterType(ClusterType.MULTI_NODE_DEDICATED_MASTER)
+@ARcaGraph(ElasticSearchAnalysisGraph.class)
+//specify a custom rca.conf to set the collector time periods to 5s to reduce runtime
+@ARcaConf(dataNode = CACHE_TUNING_RESOURCES_DIR + "rca.conf")
+@AMetric(
+        name = Cache_FieldData_Size.class,
+        dimensionNames = {
+                AllMetrics.CommonDimension.Constants.INDEX_NAME_VALUE,
+                AllMetrics.CommonDimension.Constants.SHARDID_VALUE
+        },
+        tables = {
+                @ATable(
+                        hostTag = HostTag.DATA_0,
+                        tuple = {
+                                @ATuple(
+                                        dimensionValues = {INDEX_NAME, SHARD_ID},
+                                        sum = 8500.0, avg = 8500.0, min = 8500.0, max = 8500.0)
+                        }),
+                @ATable(
+                        hostTag = {HostTag.ELECTED_MASTER},
+                        tuple = {
+                                @ATuple(
+                                        dimensionValues = {INDEX_NAME, SHARD_ID},
+                                        sum = 100.0, avg = 100.0, min = 100.0, max = 100.0)
+                        })
+        })
+@AMetric(
+        name = Cache_FieldData_Eviction.class,
+        dimensionNames = {
+                AllMetrics.CommonDimension.Constants.INDEX_NAME_VALUE,
+                AllMetrics.CommonDimension.Constants.SHARDID_VALUE
+        },
+        tables = {
+                @ATable(
+                        hostTag = HostTag.DATA_0,
+                        tuple = {
+                                @ATuple(
+                                        dimensionValues = {INDEX_NAME, SHARD_ID},
+                                        sum = 1.0, avg = 1.0, min = 1.0, max = 1.0)
+                        }),
+                @ATable(
+                        hostTag = {HostTag.ELECTED_MASTER},
```

Comment:
Same as above, please confirm and remove if this metric is missing from dedicated master scenarios. We want to make sure the integ tests simulate actual cluster scenarios, and that there are no NPEs due to missing shards.